src/index/io.rs

```diff
@@ -74,4 +127,11 @@ mod tests {
         assert_eq!(index.ntotal(), 5);
         ::std::fs::remove_file(&filepath).unwrap();
     }
+
+    #[test]
+    fn test_read_with_flags() {
+        let index = read_index_with_flags("file_name", io_flag::MEM_MAP | io_flag::READ_ONLY);
```

Comment:
Just to make it more self explanatory.
```suggestion
        let index = read_index_with_flags("file_name_that_does_not_exist", io_flag::MEM_MAP | io_flag::READ_ONLY);
```